### ISSUE TITLE
Ability to specify fresh color based on age of resources

### ISSUE THREAD

**Issue by zx8** [kubecolor/kubecolor#199]
### Description

Currently, we can specify `KUBECOLOR_OBJ_FRESH` which highlights objects in green if they are under this threshold.

It would be nice to be able to specify the color of resources based on their age, so for example:

```
KUBECOLOR_OBJ_AGE_COLORS=5m:green,2h:yellow,1d:red
```

This is just a quick example, the final implementation might be more sophisticated.

**Comment by applejag** [comment]
This is a good idea! The tricky part is to design the config of it.

I'm not too fond of throwing themeing configs into non-theming config fields. I.e, having theme configs outside of the `KUBECOLOR_THEME_...` section. But if we can't come up with anything better then maybe that's the way to go.

We already have the `KUBECOLOR_THEME_DATA_DURATIONFRESH` config which controls this. One approach would be to have the "object fresh thresholds" part in one config and then the theming in another?

Such as:

```bash
export KUBECOLOR_OBJ_FRESH="5m/2h/1d"
export KUBECOLOR_THEME_DATA_DURATIONFRESH="green/yellow/red"
```

^making use of our existing `color[]` syntax: https://kubecolor.github.io/reference/types/color-list/

Then in the case you don't specify the same number of elements you'd get:


```bash
# only uses green & yellow
export KUBECOLOR_OBJ_FRESH="5m/2h"
export KUBECOLOR_THEME_DATA_DURATIONFRESH="green/yellow/red"

# 1d gets green color as the "color[]" type cycles
export KUBECOLOR_OBJ_FRESH="5m/2h/1d"
export KUBECOLOR_THEME_DATA_DURATIONFRESH="green/yellow"

# clearer example of how it would cycle, where you'd get:
#   5m: green
#   10m: yellow
#   15m: red
#   20m: green
#   25m: yellow
#   30m: red
export KUBECOLOR_OBJ_FRESH="5m/10m/15m/20m/25m/30m"
export KUBECOLOR_THEME_DATA_DURATIONFRESH="green/yellow/red"
```

Alternatively it would make more sense if the thresholds were stretched out on to the colors, so you'd instead get this instead of repeating cycles:

```bash
# same config as above, but where we image the "cycle" logic is different:
#   5m: green
#   10m: green
#   15m: yellow
#   20m: yellow
#   25m: red
#   30m: red
export KUBECOLOR_OBJ_FRESH="5m/10m/15m/20m/25m/30m"
export KUBECOLOR_THEME_DATA_DURATIONFRESH="green/yellow/red"
```

**Comment by applejag** [comment]
I'd imagine it'd be best to just discard those colors. I can't imagine a more intuitive behaviour of what I would expect would happen.

All of these edge cases would need to be tested and documented though

**Comment by prune998** [comment]
yep, as much as I like the idea, adding too much logic and complexity will defeat the feature.
Also, if we go into it, we'll have to define a set of colors for age of resources, as we're already using green for OK, orange for warn and red for bad.

my understanding is that the number of cases in `KUBECOLOR_OBJ_FRESH` should match the number in `KUBECOLOR_THEME_DATA_DURATIONFRESH`, or we would either use the latest color for all other time, or throw an error message that the 2 should match.
